libfreerdp/core/gateway/rdg.c

```diff
@@ -298,21 +305,113 @@ static BOOL rdg_write_packet(rdpRdg* rdg, wStream* sPacket)
 	return TRUE;
 }
 
-static BOOL rdg_read_all(rdpTls* tls, BYTE* buffer, int size)
+static int rdg_socket_read(BIO* bio, BYTE* pBuffer, size_t size, UINT16 transferEncoding,
```

Comment:
In this function would be nice to:

* use early returns to avoid if then else stairs
* usually when the if statement only has one instruction we avoid brackets